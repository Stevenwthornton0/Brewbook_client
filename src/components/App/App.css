
.Header {
    width: 100%;
    margin: auto auto;
    background-color: #424242;
}

.App_main {
    background-color: #f5f5f5;
    min-height: calc(100vh - 76px);
    padding-bottom: 48px;
}