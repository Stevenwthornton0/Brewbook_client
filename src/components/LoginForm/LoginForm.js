import React, { Component } from 'react';
import TokenService from '../../services/token-service';
import AuthApiService from '../../services/auth-api-service.js';
import BreweryContext from '../../contexts/BreweryContext';
import { Button, Input } from '../../Utils/Utils';
import './LoginForm.css';

export default class LoginForm extends Component {
    static defaultProps = {
        onLoginSuccess: () => {}
    };

    static contextType = BreweryContext;

    state = { error: null };

    handleSubmitJwtAuth = ev => {
        ev.preventDefault()
        this.setState({ error: null })
        const { user_name, password } = ev.target;

        AuthApiService.postLogin({
            user_name: user_name.value,
            password: password.value
        })
            .then(res => {
                TokenService.saveUsername(user_name.value)
                user_name.value = ''
                password.value = ''
                TokenService.saveAuthToken(res.authToken)
                this.props.onLoginSuccess()
            })
            .catch(res => {
                this.setState({ error: res.error })
            })
    }

    render() {
        const { error } = this.state;
        return (
            <form
                className='LoginForm'
                onSubmit={this.handleSubmitJwtAuth}
            >
                <div role='alert'>
                    {error && <p className='error'>{error}</p>}
                </div>
                <div className='user_name'>
                    <label htmlFor='LoginForm_user_name'>
                        User Name
                    </label>
                    <Input
                        className='Input user_name'
                        required
                        name='user_name'
                        id='LoginForm_user_name'
                    />
                </div>
                <div className='password'>
                    <label htmlFor='LoginForm_password'>
                        Password
                    </label>
                    <Input
                        className='Input password'
                        required
                        name='password'
                        type='password'
                        id='LoginForm_passsword'
                    />
                </div>
                <Button type='submit'>
                    Log in
                </Button>
            </form>
        )
    }
}