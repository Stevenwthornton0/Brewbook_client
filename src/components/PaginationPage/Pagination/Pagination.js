import React, { Component } from 'react';

class Pagination extends Component {

    static defaultProps = {
        handlePagination: () => {}
    }
    
    render() {
        const { page, currentPage } = this.props

        if (page === currentPage) {
            return ( <button className='disabled' disabled={true}>{page}</button> )
        }

        return (
            <button onClick={() => {this.props.handlePagination(page)}}>{page}</button>
        )
    }
}

export default Pagination;

