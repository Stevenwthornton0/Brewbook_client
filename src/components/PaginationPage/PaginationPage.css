.paginationContainer {
    margin-top: 30px;
}

.pagination {
    display: flex;
    flex-direction: row;
    width: fit-content;
    border: 1px solid black;
    border-radius: 4px;
}

.pagination > button {
    border: none;
    padding: 4px 10px;
    border-left: 1px solid black;
    font-size: 12px;
}

.pagination > .left {
    border-left: none;
    border-radius: 4px 0 0 4px;
}

.pagination > .right {
    border-radius: 0 4px 4px 0;
}

.disabled:hover {
    color: #bdbdbd;
    background-color: white;
    cursor: default;
    box-shadow: none;
}