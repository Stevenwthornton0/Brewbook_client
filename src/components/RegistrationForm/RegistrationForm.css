.RegistrationForm {
    width: 250px;
    height: 450px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin-top: 15px;
}

label {
    display: block;
    font-size: 14px;
    font-weight: 500;
    margin-bottom: 4px;
} 

.requires {
    line-height: 8px;
    font-size: 12px;
    margin-top: 0;
    color: #424242
}
