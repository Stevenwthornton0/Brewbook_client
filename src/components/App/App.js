import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Header from '../Header/header';
import PublicOnlyRoute from '../../Utils/PublicOnlyRoute';
import PrivateOnlyRoute from '../../Utils/PrivateOnlyRoute';
import ReviewsListPage from '../../routes/ReviewsListPage/ReviewsListPage';
import LoginPage from '../../routes/LoginPage/LoginPage';
import RegistrationPage from '../../routes/RegistrationPage/RegistrationPage';
import NotFoundPage from '../../routes/NotFoundPage/NotFoundPage';
import BreweryPage from '../../routes/BreweryPage/BreweryPage';
import LoginToSee from '../../routes/LoginToSee/LoginToSee';
import FrontPage from '../../routes/FrontPage/FrontPage';
import './App.css';


class App extends React.Component {
  state = { hasError: false }

  static getDerivedStateFromError(error) {
    console.error(error)
    return { hasError: true }
  }
  
  render() {
    return (
      <div className='App'>

        <header className='Header'>
          <Header />
        </header>

        <main className='App_main'>
          <Switch>
            <Route 
              exact
              path={'/'}
              component={FrontPage}
            />

            <Route
              path={'/search'}
              component={ReviewsListPage}
            />
            <PublicOnlyRoute
              path={'/login'}
              component={LoginPage}
            />
            <PublicOnlyRoute
              path={'/loginfirst'}
              component={LoginToSee}
            />
            <PublicOnlyRoute
              path={'/register'}
              component={RegistrationPage}
            />
            <PrivateOnlyRoute 
              path={'/brewery/:breweryId'}
              component={BreweryPage}
            />

            <Route
              component={NotFoundPage}
            />
          </Switch>
        </main>
        
      </div>
    )
  }
}

export default App;