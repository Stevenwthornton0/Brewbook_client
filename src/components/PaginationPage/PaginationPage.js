import React, { Component } from 'react';
import BreweryListContext from '../../contexts/BreweryListContext';
import Pagination from './Pagination/Pagination';
import './PaginationPage.css'

class PaginationPage extends Component {
    static contextType = BreweryListContext;

    constructor(props) {
        super(props)
        this.state = {
            renderPagination: null,
            pages: []
        }  
    }

    static defaultProps = {
        handlePagination: () => {},
        handlePrev: () => {},
        handleNext: () => {},
        currentPage: 1
    }

    componentDidMount() {
        this.setState({
            renderPagination: true
        })
    }


    createPagination = () => {
        const { maxPage, currentPage } = this.props;
        const paginationPage = () => {
            const { pages } = this.context;
            return pages.map(page => 
                <Pagination
                    currentPage={this.props.currentPage} 
                    key={page}
                    page={page}
                    handlePagination={this.props.handlePagination}
                />
            )
        }
        
        if (maxPage === 1 || maxPage === 0) {
            return null
        }
        
        if (currentPage === 1) {
            return (
                <div className='pagination'>
                    <button className='left disabled' disabled={true}>prev</button>
                    {paginationPage()}
                    <button className='right' onClick={() => {this.props.handleNext()}}>next</button>
                </div>
            )

        } else if (currentPage === maxPage) {
            return (
                <div className='pagination'>
                    <button className='left' onClick={() => {this.props.handlePrev()}}>prev</button>
                    {paginationPage()}
                    <button className='right disabled' disabled={true}>next</button>
                </div>
            )
        }

        return (
            <div className='pagination'>
                <button className='left' onClick={() => {this.props.handlePrev()}}>prev</button>
                {paginationPage()}
                <button className='right' onClick={() => {this.props.handleNext()}}>next</button>
            </div>
        )

    }
    
    render() {
        return (
            <div className='paginationContainer'>
                {this.state.renderPagination ? this.createPagination() : null}
            </div>
        )
    }
}

export default PaginationPage;

// creates a pagination based on the amount of pages and responses to a search