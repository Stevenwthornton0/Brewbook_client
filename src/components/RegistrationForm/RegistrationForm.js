import React, { Component } from 'react';
import AuthApiService from '../../services/auth-api-service';
import TokenService from '../../services/token-service';
import LoaderSpinner from '../../Utils/LoaderSpinner';
import { Input, Button } from '../../Utils/Utils';
import './RegistrationForm.css';

export default class RegistrationForm extends Component {
    static defaultProps = {
        onRegistrationSuccess: () => {}
    }

    state = { 
        error: null,
        waiting: false 
    };

    
    toSnakeCase = function(string) {
      var s;
      s = string.replace(/[^\w\s]/g, "");
      s = s.replace(/\s+/g, " ");
      return s.toLowerCase().split(' ').join('_');
    };

    handleSubmit = ev => {
        ev.preventDefault()
        const { first_name, last_name, user_name, password } = ev.target;

        this.setState({ error: null, waiting: true });
        AuthApiService.postUser({
            first_name: first_name.value,
            last_name: last_name.value,
            user_name: user_name.value,
            password: password.value,
            admin: 'false'
        })
            .then(user => {
                first_name.value = ''
                last_name.value = ''
                user_name.value = ''
                password.value = ''
                this.props.onRegistrationSuccess()
            })
            .catch(res => {
                this.setState({error: res.error })
            })
    }

    render() {
        const { error } = this.state;
        return (
            <form
                className='RegistrationForm'
                onSubmit={this.handleSubmit}
            >
                <div role='alert'>
                    {error && <p>{error}</p>}
                </div>
                <div className='first_name'>
                    <label htmlFor='RegistrationForm_first-name'>
                        First Name
                    </label>
                    <Input
                        name='first_name'
                        type='text'
                        required
                        id='RegistrationForm_first-name'
                    />
                </div>
                <div className='last_name'>
                    <label htmlFor='RegistrationForm_last-name'>
                        Last Name
                    </label>
                    <Input
                        name='last_name'
                        type='text'
                        required
                        id='RegistrationForm_last-name'
                    />
                </div>
                <div className='user_name'>
                    <label htmlFor='RegistrationForm_user-name'>
                        User Name
                    </label>
                    <Input
                        name='user_name'
                        type='text'
                        required
                        id='RegistrationForm_user-name'
                    />
                </div>
                <div className='password'>
                    <label htmlFor='RegistrationForm_password'>
                        Password
                    </label>
                    <Input
                        name='password'
                        type='password'
                        required
                        id='RegistrationForm_password'
                    />
                </div>
                <div className='requires'>
                    <p className='requires'>Requirements:</p>
                    <p>&bull; an uppercase letter</p>
                    <p>&bull; a lowercase letter</p>
                    <p>&bull; a number</p>
                    <p>&bull; a special character</p>
                </div>
                <Button type='submit'>
                    Register
                </Button>
                <div className='loaderContainer'>
                    {this.state.waiting ? <LoaderSpinner /> : null}
                </div>
            </form>
        )
    }
}