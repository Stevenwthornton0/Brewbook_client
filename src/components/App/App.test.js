import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';
import Header from '../Header/header';
import ReviewsListPage from '../../routes/ReviewsListPage/ReviewsListPage';
import LoginPage from '../../routes/LoginPage/LoginPage';
import LoginToSee from '../../routes/LoginToSee/LoginToSee';
import RegistrationPage from '../../routes/RegistrationPage/RegistrationPage';
import BreweryPage from '../../routes/BreweryPage/BreweryPage';
import LoginForm from '../LoginForm/LoginForm';
import PaginationPage from '../PaginationPage/PaginationPage';
import Pagination from '../PaginationPage/Pagination/Pagination';
import RegistrationForm from '../RegistrationForm/RegistrationForm';
import ReviewForm from '../ReviewForm/ReviewForm';
import SearchBar from '../SearchBar/SearchBar';
import SearchResults from '../SearchBar/SearchResults/SearchResults';
import BreweryList from '../SearchBar/SearchResults/BreweryList/BreweryList';
import BreweryItem from '../SearchBar/SearchResults/BreweryList/BreweryItem/BreweryItem';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'

Enzyme.configure({ adapter: new Adapter() })

describe('All routes', () => {
    it('App renders without crashing', () => {
        const wrapper = shallow(<App />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('Header renders without crashing', () => {
        const wrapper = shallow(<Header />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('ReviewsListPage renders without crashing', () => {
        const wrapper = shallow(<ReviewsListPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('LoginToSee renders without crashing', () => {
        const wrapper = shallow(<LoginToSee />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('RegistrationPage renders without crashing', () => {
        const wrapper = shallow(<RegistrationPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('BreweryPage renders without crashing', () => {
        const wrapper = mount(<BreweryPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('LoginPage renders without crashing', () => {
        const wrapper = shallow(<LoginPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})

describe('all components', () => {
    it('LoginForm renders without crashing', () => {
        const wrapper = shallow(<LoginForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('PaginationPage renders without crashing', () => {
        const wrapper = mount(<PaginationPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('Pagination renders without crashing', () => {
        const wrapper = shallow(<Pagination pages={[]}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('RegistrationForm renders without crashing', () => {
        const wrapper = shallow(<RegistrationForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('ReviewForm renders without crashing', () => {
        const wrapper = shallow(<ReviewForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('SearchBar renders without crashing', () => {
        const wrapper = shallow(<SearchBar />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('SearchResults renders without crashing', () => {
        const wrapper = shallow(<SearchResults />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('BreweryList renders without crashing', () => {
        const wrapper = mount(<BreweryList />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('BreweryItem renders without crashing', () => {
        const brewery = {
            id: 6780,
            name: "Jester King Brewery",
            brewery_type: "micro",
            street: "13187 Fitzhugh Rd",
            city: "Austin",
            state: "Texas",
            postal_code: "78736-6510",
            country: "United States",
            longitude: "-98.0824692",
            latitude: "30.2547264",
            phone: "5125375100",
            website_url: "http://www.jesterkingbrewery.com",
            updated_at: "2018-08-24T16:21:26.875Z"
            }
        const wrapper = shallow(<BreweryItem brewery={brewery}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})