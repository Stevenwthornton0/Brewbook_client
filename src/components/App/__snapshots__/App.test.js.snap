// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All routes App renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="Header"
  >
    <Header />
  </header>
  <main
    className="App_main"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <PublicOnlyRoute
        component={[Function]}
        path="/login"
      />
      <PublicOnlyRoute
        component={[Function]}
        path="/loginfirst"
      />
      <PublicOnlyRoute
        component={[Function]}
        path="/register"
      />
      <PrivateOnlyRoute
        component={[Function]}
        path="/brewery/:breweryId"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </main>
</div>
`;

exports[`All routes BreweryPage renders without crashing 1`] = `
<BreweryPage
  match={
    Object {
      "params": Object {},
    }
  }
>
  <div>
    <div
      className="breweryPage"
    >
      <h1 />
      <div
        className="contact-container"
      />
      <h2>
        Reviews
      </h2>
      <breweryReviews
        deleteReview={[Function]}
        reviews={Array []}
      >
        <p>
          No one has posted a review
        </p>
      </breweryReviews>
      <ReviewForm>
        <div>
          <h3>
            Post a Review
          </h3>
          <form
            className="ReviewForm"
            onSubmit={[Function]}
          >
            <label>
              Rating
            </label>
            <select
              className="rating"
              id="rating"
            >
              <option
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
              <option
                value="3"
              >
                3
              </option>
              <option
                value="4"
              >
                4
              </option>
              <option
                value="5"
              >
                5
              </option>
            </select>
            <label>
              Review
            </label>
            <Textarea
              aria-label="Leave a review..."
              className="reviewText"
              cols="30"
              id="text"
              placeholder="Leave a review.."
              required={true}
              rows="4"
            >
              <textarea
                aria-label="Leave a review..."
                className="Textarea reviewText"
                cols="30"
                id="text"
                placeholder="Leave a review.."
                required={true}
                rows="4"
              />
            </Textarea>
            <Button
              type="submit"
            >
              <button
                className="Button "
                type="submit"
              >
                Post
              </button>
            </Button>
          </form>
        </div>
      </ReviewForm>
    </div>
  </div>
</BreweryPage>
`;

exports[`All routes Header renders without crashing 1`] = `
<nav
  className="header"
>
  <div
    className="header_container"
  >
    <Link
      className="logo"
      to="/"
    >
      <h1>
        <FontAwesomeIcon
          border={false}
          className="logo"
          fixedWidth={false}
          flip={null}
          icon="beer"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
         
        brewbook
      </h1>
    </Link>
    <div
      className="header_logged-out"
    >
      <Link
        to="/register"
      >
        Register
      </Link>
      <Hyph />
      <Link
        to="/login"
      >
        Login
      </Link>
    </div>
  </div>
</nav>
`;

exports[`All routes LoginPage renders without crashing 1`] = `
<section
  className="loginPage"
>
  <h2>
    Login
  </h2>
  <LoginForm
    onLoginSuccess={[Function]}
  />
</section>
`;

exports[`All routes LoginToSee renders without crashing 1`] = `
<section
  className="loginPage"
>
  <h2>
    Login 
    <span
      className="smaller"
    >
      to view full brewery details
    </span>
  </h2>
  <LoginForm
    onLoginSuccess={[Function]}
  />
</section>
`;

exports[`All routes RegistrationPage renders without crashing 1`] = `
<section
  className="registrationPage"
>
  <h2>
    Register
  </h2>
  <RegistrationForm
    onRegistrationSuccess={[Function]}
  />
</section>
`;

exports[`All routes ReviewsListPage renders without crashing 1`] = `
<div
  className="searchForm"
>
  <h2>
    Search for a brewery
  </h2>
  <SearchBar />
</div>
`;

exports[`all components BreweryItem renders without crashing 1`] = `
<Link
  className="brewery"
  to="/brewery/6780"
>
  <section
    className="breweryItem"
  >
    <h3>
      Jester King Brewery
    </h3>
    <div
      className="container1"
    >
      <div
        className="container2"
      >
        <p>
          13187 Fitzhugh Rd
        </p>
        <p>
          Austin
          , 
          Texas
        </p>
      </div>
      <div
        className="container3"
      >
        <p
          className="brewery website"
        >
          <FontAwesomeIcon
            border={false}
            className="logo"
            fixedWidth={false}
            flip={null}
            icon="mouse"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
          <span
            className="url"
          >
            jesterkingbrewery.com
          </span>
        </p>
        <p
          className="brewery phone"
        >
          <FontAwesomeIcon
            border={false}
            className="logo"
            fixedWidth={false}
            flip={null}
            icon="phone"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
           
          512-537-5100
        </p>
      </div>
    </div>
  </section>
</Link>
`;

exports[`all components BreweryList renders without crashing 1`] = `
<BreweryList>
  <div />
</BreweryList>
`;

exports[`all components LoginForm renders without crashing 1`] = `
<div>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="user_name"
    >
      <label
        htmlFor="LoginForm_user_name"
      >
        User Name
      </label>
      <Input
        className="Input user_name"
        id="LoginForm_user_name"
        name="user_name"
        required={true}
      />
    </div>
    <div
      className="password"
    >
      <label
        htmlFor="LoginForm_password"
      >
        Password
      </label>
      <Input
        className="Input password"
        id="LoginForm_passsword"
        name="password"
        required={true}
        type="password"
      />
    </div>
    <Button
      type="submit"
    >
      Log in
    </Button>
  </form>
  <div
    className="loaderContainer"
  />
</div>
`;

exports[`all components Pagination renders without crashing 1`] = `
<button
  className="disabled"
  disabled={true}
/>
`;

exports[`all components PaginationPage renders without crashing 1`] = `
<PaginationPage
  currentPage={1}
  handleNext={[Function]}
  handlePagination={[Function]}
  handlePrev={[Function]}
>
  <div
    className="paginationContainer"
  >
    <div
      className="pagination"
    >
      <button
        className="left disabled"
        disabled={true}
      >
        prev
      </button>
      <button
        className="right"
        onClick={[Function]}
      >
        next
      </button>
    </div>
  </div>
</PaginationPage>
`;

exports[`all components RegistrationForm renders without crashing 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="first_name"
  >
    <label
      htmlFor="RegistrationForm_first-name"
    >
      First Name
    </label>
    <Input
      id="RegistrationForm_first-name"
      name="first_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="last_name"
  >
    <label
      htmlFor="RegistrationForm_last-name"
    >
      Last Name
    </label>
    <Input
      id="RegistrationForm_last-name"
      name="last_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm_user-name"
    >
      User Name
    </label>
    <Input
      id="RegistrationForm_user-name"
      name="user_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm_password"
    >
      Password
    </label>
    <Input
      id="RegistrationForm_password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <div
    className="requires"
  >
    <p
      className="requires"
    >
      Requirements:
    </p>
    <p>
      • an uppercase letter
    </p>
    <p>
      • a lowercase letter
    </p>
    <p>
      • a number
    </p>
    <p>
      • a special character
    </p>
  </div>
  <Button
    type="submit"
  >
    Register
  </Button>
</form>
`;

exports[`all components ReviewForm renders without crashing 1`] = `
<div>
  <h3>
    Post a Review
  </h3>
  <form
    className="ReviewForm"
    onSubmit={[Function]}
  >
    <label>
      Rating
    </label>
    <select
      className="rating"
      id="rating"
    >
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
      <option
        value="3"
      >
        3
      </option>
      <option
        value="4"
      >
        4
      </option>
      <option
        value="5"
      >
        5
      </option>
    </select>
    <label>
      Review
    </label>
    <Textarea
      aria-label="Leave a review..."
      className="reviewText"
      cols="30"
      id="text"
      placeholder="Leave a review.."
      required={true}
      rows="4"
    />
    <Button
      type="submit"
    >
      Post
    </Button>
  </form>
</div>
`;

exports[`all components SearchBar renders without crashing 1`] = `
<div
  className="mainContainer"
>
  <div
    className="formContainer"
  >
    <form
      className="SearchForm"
      onSubmit={[Function]}
    >
      <div
        className="city"
      >
        <label
          htmlFor="SearchForm_city"
        >
          City
        </label>
        <Input
          id="SearchForm_city"
          name="city"
          required={true}
          type="text"
        />
      </div>
      <div
        className="state"
      >
        <label
          htmlFor="SearchForm_state"
        >
          State
        </label>
        <States />
      </div>
      <Button
        type="submit"
      >
        Search
      </Button>
    </form>
  </div>
</div>
`;

exports[`all components SearchResults renders without crashing 1`] = `
<div>
  <div
    className="SearchResults"
  >
    <h2>
      0
       Results for 
      , 
    </h2>
    <div
      className="breweryContainer"
    >
      <BreweryList />
    </div>
  </div>
</div>
`;
